File: BasicSampleJava/app/src/main/java/com/regula/facesample/items/customization/fragment/LivenessProcessingCustomFragment.java
package com.regula.facesample.items.customization.fragment;

import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.regula.facesample.R;
import com.regula.facesdk.fragment.BaseFaceLivenessProcessingFragment;

/**
 * Created by Sergey Yakimchik on 04/29/22.
 * Copyright (c) 2020 Regula. All rights reserved.
 */

public final class LivenessProcessingCustomFragment extends BaseFaceLivenessProcessingFragment {

    private TextView guidelinesTextView;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = super.onCreateView(inflater, container, savedInstanceState);
        if (view == null)
            return null;

        guidelinesTextView = view.findViewById(R.id.guidelines_text_view);

        return view;
    }

    @Override
    public int getResourceLayoutId() {
        return R.layout.custom_liveness_processing;
    }

    @Override
    public View getRetryButton(View view) {
        return view.findViewById(R.id.retry_button);
    }

    @Override
    public View getCancelButton(View view) {
        return view.findViewById(R.id.cancel_button);
    }

    @Override
    public void faceSdkError(int[] stringResourceId) {
        StringBuilder guidelineTextBuilder = new StringBuilder();
        if (stringResourceId == null) {
            stringResourceId = new int[] { com.regula.facesdk.R.string.livenessRetry_text_environment, com.regula.facesdk.R.string.livenessRetry_text_subject };
            Log.e("CustomFragment", "Not defined error caught");
        }

        for (int strId : stringResourceId) {
            guidelineTextBuilder.append(String.format("- %s\n", getString(strId)));
        }

        guidelinesTextView.setText(guidelineTextBuilder);
    }

    @Override
    public View getProcessingLayout(View v) {
        return v.findViewById(R.id.processing_layout);
    }

    @Override
    public View getRetryLayout(View v) {
        return v.findViewById(R.id.result_layout);
    }

    @Override
    public View getVerifyLayout(View view) {
        return view.findViewById(R.id.verifiedLayout);
    }

    @Override
    public View getCloseButton(View v) {
        return v.findViewById(R.id.close_button);
    }

}

File: FaceSampleKotlin/app/src/main/java/com/regula/facesamplekotlin/fragments/LivenessProcessingCustomFragment.kt
package com.regula.facesamplekotlin.fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.regula.common.utils.RegulaLog
import com.regula.facesamplekotlin.R
import com.regula.facesdk.fragment.BaseFaceLivenessProcessingFragment


class LivenessProcessingCustomFragment : BaseFaceLivenessProcessingFragment() {
    private var guidelinesTextView: TextView? = null
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val view: View = super.onCreateView(inflater, container, savedInstanceState)
            ?: return null
        guidelinesTextView = view.findViewById(R.id.guidelines_text_view)
        return view
    }

    override fun getResourceLayoutId(): Int {
        return R.layout.custom_liveness_processing;
    }

    override fun getRetryButton(view: View): View {
        return view.findViewById(R.id.retry_button)
    }

    override fun getCancelButton(view: View): View {
        return view.findViewById(R.id.cancel_button)
    }

    override fun faceSdkError(stringResourceId: IntArray) {
        var stringResourceId: IntArray? = stringResourceId
        val guidelineTextBuilder = StringBuilder()
        if (stringResourceId == null) {
            stringResourceId = intArrayOf(
                R.string.livenessRetry_text_environment,
                R.string.livenessRetry_text_subject
            )
            RegulaLog.e("Not defined error caught")
        }
        for (strId in stringResourceId) {
            guidelineTextBuilder.append(String.format("- %s\n", getString(strId)))
        }
        guidelinesTextView?.text = guidelineTextBuilder
    }

    override fun getProcessingLayout(v: View): View {
        return v.findViewById(R.id.processing_layout)
    }

    override fun getRetryLayout(v: View): View {
        return v.findViewById(R.id.result_layout)
    }

    override fun getCloseButton(v: View): View {
        return v.findViewById(R.id.close_button)
    }
}